### PR TITLE
cgo pkg-config fixes

### PR DESCRIPTION
**Pull request by sheenobu**
This fixes two issues with cgo support:

 * In go 1.4, the pkg-config values must be passed to cgo after the -- (might be the same for go 1.5). CGO_CFLAGS seems to be ignored.
 * Both versions require the cflags/pkg-config to be passed to rungcc1.

Repo which replicates this issue with working output vs current gb master output: https://github.com/sheenobu/gb-cgo-bug

### CODE REVIEW

File: cgo14.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ func cgo(pkg *Package) ([]ObjTarget, []string) {
 	cgoCFLAGS = append(cgoCFLAGS, pcCFLAGS...)
 	cgoLDFLAGS = append(cgoLDFLAGS, pcLDFLAGS...)
 
-	if err := runcgo1(pkg, cgoCFLAGS, cgoLDFLAGS); err != nil {
+	if err := runcgo1(pkg, cgoCFLAGS, pcCFLAGS, cgoLDFLAGS); err != nil {
```
Comment (davecheney) [inline cgo14.go]: Why is this necessary, pcFLAGS are already included in cgoCFLAGS

**Comment by davecheney** [comment]
Thanks for your fix and the reproduction, i'll include this integration tests.

File: cgo14.go (hunk 1 above)
Comment (sheenobu) [inline cgo14.go]: cgo is not respecting the contents of cgoCFLAGS. Additionally, cgoCFLAGS is CFLAGS + "-g -02" + pcCFLAGS. -g -02 are not valid arguments for cgo which means we need to include pcCFLAGS separately when building the full cgo command.

**Comment by sheenobu** [comment]
@davecheney Great, thank you.

**Comment by davecheney** [comment]
@sheenobu please take a look at the integration test failures.

**Comment by davecheney** [comment]
@sheenobu i'm going to take a look at this now. I've created an integration test for sdl2 and can see the problem that we're not getting the right flags out of pkg-config into runcgo1

**Comment by sheenobu** [comment]
Looks like only half of this PR was correct. Replaced by #304
